### ISSUE TITLE
False performance alerts get triggered during night time

### ISSUE THREAD

**Issue by cseas** [getsentry/sentry#53577]
### Environment

SaaS (https://sentry.io/)

### Steps to Reproduce

1. Set a p75 FCP performance alert with critical threshold as 3 seconds.
2. Alert interval set to 4 hours.
3. Wait for 24 hours. The false alert will trigger likely during night time.

### Expected Result

There should be no alert if all performance numbers are within threshold.

### Actual Result

An alert gets triggered with a negative duration.

<img width="910" alt="Screenshot 2023-07-26 at 4 15 01 AM" src="https://github.com/getsentry/sentry/assets/29686866/43342a3e-d793-4fcd-ba76-9565342a1f88">

The transactions table on the alert page also shows that all p75 FCP measurements are within the threshold so there's nothing actionable from the alert.

### Product Area

Performance

**Comment by roggenkemper** [comment]
Do you mind sending a link to the alert where you are seeing this? You can respond with it here or feel free to email it to me at richard.roggenkemper@sentry.io

**Comment by cseas** [comment]
Sent the alert link on your mail. Looks like the negative duration got corrected automatically but that wasn't the primary motivation for creating this issue.

You can see that the critical alert got triggered with a threshold of 3000ms. But when we check the transactions table below it, you can see that all of our transactions have a p75 FCP below that.

These false alerts only trigger during night time and in large numbers. We suspect that this might be happening because our user traffic is low at night so the alert gets triggered even if only 1 or 2 users face a low FCP that could be attributed to their own device's network issues.

We want to reduce these false alerts as we don't see anything actionable from these but also want to keep the alert interval low so our team gets alerted fast when something goes wrong. 1 day interval reduces false alerts but would alert us too late. 4 hour interval works according to what we want to monitor sends us a lot of false alerts.

I think if we can set something like "send alert only if this number of users are affected" or "send alert only if user traffic is at a minimum set number", that would help us remove these false alerts we get at night.

**Comment by cseas** [comment]
Thanks for confirming! We'll try out the % change tracking and see if it helps. 

While it might solve the false alerts issue, it might get difficult to track performance numbers as an overview because with our current setup, we can confidently say "all our paths have an FCP below x".

For example, if a path currently has a steady LCP of, say 6 seconds, we'd want that to show up in the alert but with % change that might get missed. Will need to discuss with the team internally on whether we can use the % change format. Our use-case of performance alerting isn't really "alert us when something gets worse by x%" but is "we have these target performance numbers, alert us when these are degraded".

I think we have the information needed from the Sentry team's end. Can this ticket be considered a feature request for adding the "send alert if X users are affected" feature?

**Comment by roggenkemper** [comment]
Yes! Let me add that information to the ticket!

**Comment by rene-leanix** [comment]
> Unfortunately, we don't have a current way to only send an alert if X users are affected, but that is a valid idea.

@roggenkemper, we have encountered the same issue: the alerts on crash-free user or session rates are too unreliable to be used for critical alerting, as long as it's not possible to also provide a minimum absolute number of users or sessions affected.

We have enabled crash-free user alerts for all our micro-front-ends. Some of those have thousands of users per day, but others only a handful. For the latter, the alerts on crash-free user or session rates trigger too easily if 1 or 2 users encounter an error. I'm trying to reduce this flakiness by increasing `timeWindow` and `thresholdPeriod` and decreasing `sensitivity`, but this can lead to the alert triggering too late when it's actually needed.

I think that being able to add minimum absolute number of users or sessions affected in addition to these other properties would help. For us it would be enough to make this available via the API.

I assume that that's not already possible by adjusting the `aggregate` or `query` properties. Is that correct?

**Comment by malwilley** [comment]
@rene-leanix you're correct, it's unfortunately not currently possible to add a minimum threshold for crash free session/user rates.

There is a similar ticket here that you can track for that functionality, though as of now that work has not been prioritized: https://github.com/getsentry/sentry/issues/66988
